src/daemon/metric.c

```diff
@@ -207,70 +245,70 @@ int metric_reset(metric_t *m) {
   }
 
   label_set_reset(&m->label);
+  label_set_reset(&m->resource);
   meta_data_destroy(m->meta);
 
   memset(m, 0, sizeof(*m));
 
   return 0;
 }
 
+static int format_label_set(strbuf_t *buf, label_set_t const *labels,
+                            char const *prefix, bool first_label) {
+  int status = 0;
+  for (size_t i = 0; i < labels->num; i++) {
+    if (!first_label) {
+      status = status || strbuf_print(buf, ",");
+    }
+
+    status = status || strbuf_print(buf, prefix);
+    status = status || strbuf_print(buf, labels->ptr[i].name);
+    status = status || strbuf_print(buf, "=\"");
+    status = status || strbuf_print_escaped(buf, labels->ptr[i].value,
+                                            "\\\"\n\r\t", '\\');
+    status = status || strbuf_print(buf, "\"");
+    first_label = false;
+  }
+  return status;
+}
+
 int metric_identity(strbuf_t *buf, metric_t const *m) {
   if ((buf == NULL) || (m == NULL) || (m->family == NULL)) {
     return EINVAL;
   }
 
   int status = strbuf_print(buf, m->family->name);
-  if (m->label.num == 0) {
+  if (m->resource.num == 0 && m->label.num == 0) {
     return status;
   }
 
   status = status || strbuf_print(buf, "{");
-  for (size_t i = 0; i < m->label.num; i++) {
-    if (i != 0) {
-      status = status || strbuf_print(buf, ",");
-    }
 
-    status = status || strbuf_print(buf, m->label.ptr[i].name);
-    status = status || strbuf_print(buf, "=\"");
-    status = status || strbuf_print_escaped(buf, m->label.ptr[i].value,
-                                            "\\\"\n\r\t", '\\');
-    status = status || strbuf_print(buf, "\"");
-  }
+  bool first_label = true;
+  status = status || format_label_set(buf, &m->resource, RESOURCE_LABEL_PREFIX,
+                                      first_label);
+
+  first_label = (m->resource.num == 0);
```

Comment:
IMHO this is odd way of writing:
```suggestion
  if (m->resource.num == 0) {
    status = status || format_label_set(buf, &m->resource, RESOURCE_LABEL_PREFIX,
                                        first_label);
    first_label = false;
  }
```